src/pod.ts

```diff
@@ -722,3 +782,581 @@ export interface HostAlias {
+
+    const topology = options.topology ?? Topology.HOSTNAME;
+    const term = this.createPodAffinityTerm(topology, selection);
+
+    if (options.weight) {
+      this.validateWeight(options.weight);
+      this._podAffinityPreferred.push({ weight: options.weight, podAffinityTerm: term });
+    } else {
+      this._podAffinityRequired.push(term);
+    }
+  }
+
+  /**
+   * Seperate this pod from a scheduling selection.
+   *
+   * A selection can be one of:
+   *
+   * - An instance of a `Pod`.
+   * - An instance of a `Workload` (e.g `Deployment`, `StatefulSet`).
+   * - An un-managed pod that can be selected via `Pod.select()`.
+   *
+   * Seperating from multiple selections acts as an AND condition. meaning the pod
+   * will not be assigned to a node that satisfies all selections (i.e runs at least one pod that satisifies each selection).
+   *
+   * Under the hood, this method utilizes the `podAntiAffinity` property.
+   *
+   * @see https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity
+   */
+  public separate(selectable: IPodSchedulingSelection, options: PodSchedulingSeparateOptions = {}) {
+
+    const topology = options.topology ?? Topology.HOSTNAME;
+    const term = this.createPodAffinityTerm(topology, selectable);
+
+    if (options.weight) {
+      this.validateWeight(options.weight);
+      this._podAntiAffinityPreferred.push({ weight: options.weight, podAffinityTerm: term });
+    } else {
+      this._podAntiAffinityRequired.push(term);
+    }
+
+  }
+
+  private createPodAffinityTerm(topology: Topology, selection: IPodSchedulingSelection): k8s.PodAffinityTerm {
+    const selector = selection.podSelector;
+    return {
+      topologyKey: topology.key,
+      labelSelector: selector.labelSelector ? {
+        matchExpressions: selector.labelSelector.map(s => ({ key: s.key, operator: s.operator!, values: s.values })),
+      } : undefined,
+      namespaceSelector: selector.namespaceSelector ? {
+        matchExpressions: selector.namespaceSelector.map(s => ({ key: s.key, operator: s.operator!, values: s.values })),
+      } : undefined,
+    };
+  }
+
+  private createNodeAffinityTerm(node: SelectedNode): k8s.NodeSelectorTerm {
+    return { matchExpressions: node.labelSelector.map(s => ({ key: s.key, operator: s.operator!, values: s.values })) };
+  }
+
+  private validateWeight(weight: number) {
+    if (weight < 1 || weight > 100) {
+      // https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity-weight
+      throw new Error(`Invalid affinity weight: ${weight}. Must be in range 1-100`);
+    }
+  }
+
+  /**
+   * @internal
+   */
+  public _toKube(): { affinity?: k8s.Affinity; nodeName?: string; tolerations?: k8s.Toleration[] } {
+
+    const atLeastOne = (...arrays: Array<any>[]) => {
+      return [...arrays].map(a => a.length).reduce((a, b) => a + b) > 0;
```

Comment:
(optional) Another way to shave the yak 😜 
```suggestion
      return arrays.flat().length > 0;
```

Reply:
Oh nice, much obliged!